homeassistant/components/pglab/sensor.py

```diff
@@ -0,0 +1,119 @@
+"""Sensor for PG LAB Electronics."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+
+from pypglab.const import SENSOR_REBOOT_TIME, SENSOR_TEMPERATURE, SENSOR_VOLTAGE
+from pypglab.device import Device as PyPGLabDevice
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import Platform, UnitOfElectricPotential, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.util.dt import utcnow
+
+from . import PGLABConfigEntry
+from .device_sensor import PGLabDeviceSensor
+from .discovery import PGLabDiscovery
+from .entity import PGLabEntity
+
+PARALLEL_UPDATES = 0
+
+SENSOR_INFO: list[SensorEntityDescription] = [
+    SensorEntityDescription(
+        key=SENSOR_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_VOLTAGE,
+        translation_key="mpu_voltage",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_REBOOT_TIME,
+        translation_key="runtime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        icon="mdi:progress-clock",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: PGLABConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up sensor for device."""
+
+    @callback
+    def async_discover(
+        pglab_device: PyPGLabDevice,
+        pglab_device_sensor: PGLabDeviceSensor,
+    ) -> None:
+        """Discover and add a PG LAB Sensor."""
+        pglab_discovery = config_entry.runtime_data
+        for description in SENSOR_INFO:
+            pglab_sensor = PGLabSensor(
+                pglab_discovery, pglab_device, pglab_device_sensor, description
+            )
+            async_add_entities([pglab_sensor])
```

Comment:
Don't call `async_add_entities` in a loop. Collect all entities first in a list and call `async_add_entities` once.